test/size_test.go

```diff
@@ -0,0 +1,192 @@
+package tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	//"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/fsouza/go-dockerclient"
+	"github.com/kr/pretty"
+	"github.com/stretchr/testify/assert"
+)
+
+type rockerBuildFn func(string, io.Writer, ...string) error
+
+func runImageSizeTestWithDockerVersion(t *testing.T, dockerVersion string) {
+
+	rocker, cleanup, err := setupDockerVersionedEnv(t, dockerVersion)
+	assert.Nil(t, err, "setup env")
+
+	defer cleanup()
+
+	time.Sleep(2 * 1e9)
+
+	fmt.Fprintf(os.Stdout, "docker env ready, building test image\n")
+
+	rockerFile := `
+FROM busybox:latest
+RUN dd if=/dev/zero of=/root/binary1M-1 bs=1 count=1000000
+TAG tag1
+RUN dd if=/dev/zero of=/root/binary1M-2 bs=1 count=1000000
+TAG tag2`
+
+	rd, wr := io.Pipe()
+	jsonRd := json.NewDecoder(rd)
+
+	result := make(chan []int)
+
+	go func() {
+		deltas := []int{}
+		for {
+			m := map[string]interface{}{}
+			if err := jsonRd.Decode(&m); err != nil {
+				if err == io.EOF {
+					break
+				}
+				fmt.Printf("decode error: %s", err)
```

Comment:
Kind of `t.Fail` would fit better here, otherwise in case of error this test will hang at line number 76